kaska/s2_observations.py

```diff
@@ -238,67 +337,56 @@ def read_granule(self, timestep):
                        axis=0) & (~mask)
         mask = ~mask1
+        if mask.sum() == 0:
+            LOG.info(f"{str(timestep):s} -> No clear observations")
+            return None, None, None, None, None, None
         rho_surface = rho_surface / 10000.0
-        
-        #mask = np.all(rho_surface > 0, axis=0) & (~mask)
-
+        rho_surface[:, mask1] = np.nan
+        # uncertainty
         rho_unc = np.array(rho_unc) / 10000.0
         rho_unc[:, mask] = np.nan
         # Average uncertainty over the image
         rho_unc = np.nanmean(rho_unc, axis=(1, 2))
-        rho_surface[:, mask1] = np.nan
-        if mask.sum() == 0:
-            LOG.info(f"{str(timestep):s} -> No clear observations")
-            return None, None, None, None, None, None
         LOG.info(
             f"{str(timestep):s} -> Total of {mask.sum():d} clear pixels "
             + f"({100.*mask.sum()/np.prod(mask.shape):f}%)"
         )
-        # Now read angles
-        sun_angles = reproject_data(
-            str(current_folder.parent / "ANG_DATA/SAA_SZA.tif"),
-            target_img=self.state_mask,
-            xRes=20,
-            yRes=20,
-            resample=0,
-        ).ReadAsArray()
-        view_angles = reproject_data(
-            str(current_folder.parent / "ANG_DATA/VAA_VZA_B05.tif"),
-            target_img=self.state_mask,
-            xRes=20,
-            yRes=20,
-            resample=0,
-        ).ReadAsArray()
-        sza = np.cos(np.deg2rad(sun_angles[1].mean() / 100.0))
-        vza = np.cos(np.deg2rad(view_angles[1].mean() / 100.0))
-        saa = sun_angles[0].mean() / 100.0
-        vaa = view_angles[0].mean() / 100.0
-        raa = np.cos(np.deg2rad(vaa - saa))
-
-        return rho_surface, mask, sza, vza, raa, rho_unc
-
 
-if __name__ == "__main__":
-    time_grid = []
-    today = dt.datetime(2017,1,1)
-    while (today <= dt.datetime(2017, 12, 31)):
-        time_grid.append(today)
-        today += dt.timedelta(days=5)
+        # Read in angles
+        sza = None
+        vza = None
+        raa = None
+        if ((current_folder/"ANG_DATA/SAA_SZA.tif").exists() and
+            (current_folder/"ANG_DATA/VAA_VZA_B05.tif").exists()):
+            sun_angles = reproject_data(
+                str(current_folder / "ANG_DATA/SAA_SZA.tif"),
+                target_img=self.state_mask,
+                xRes=20,
+                yRes=20,
+                resample=0,
+            ).ReadAsArray()
+            view_angles = reproject_data(
+                str(current_folder / "ANG_DATA/VAA_VZA_B05.tif"),
+                target_img=self.state_mask,
+                xRes=20,
+                yRes=20,
+                resample=0,
+            ).ReadAsArray()
+            sza = np.cos(np.deg2rad(sun_angles[1].mean() / 100.0))
+            vza = np.cos(np.deg2rad(view_angles[1].mean() / 100.0))
+            saa = sun_angles[0].mean() / 100.0
+            vaa = view_angles[0].mean() / 100.0
```

Comment:
A comment saying what's the first and second indice of the view and sun angles woudl be helpul to know what's going on here.

Reply:
I don't know the answer to that. @jgomezdans ?

Reply:
Trained the neural nets with the cosines.
za = zenith angle. 90 - Altitude 
v  is sensor view
s is sun.
In 100s because: 36º34 is stored as 3634.